src/commands.ts

```diff
@@ -72,7 +82,61 @@ export class CommandHandler {
 
         return true;
     }
+    
+
+    /**
+     * Register a new command
+     * @param client - tmi client
+     * @param message - the chat message that called the command
+     * @param tags - the chatters tags which determine permissions
+     * @param channel - the channel the bot will run the command in
+     * @returns success status
+     */
+    processCommand(client: tmi.Client, message: string, tags: tmi.ChatUserstate, channel: string): boolean {
+        if(!message.startsWith(this.options.prefix)) {
+            return false
+        }
+
+        // extract command and arguments
+        const partsOfCommand = message.slice(this.options.prefix.length).split(/\s+/)
+        const commandName = partsOfCommand[0].toLowerCase() // command name is only the first element
+        // TODO: handle additional arguments in commands
+
+        if(!this.commands.has(commandName)) {
+            client.say(channel, "Invalid Command")
+            return false
+        }
+
+        if(!this.checkPermissions(tags, this.options.requiredPermissions || ["broadcaster"])) {
+            client.say(channel, "Permission Denied")
+            return false
+        }
+
+        const command = this.commands.get(commandName) // retrieves command (includes handler and options)
+        // TODO: actually run commands
+
+        return true
+    }
 
 
+    checkPermissions(tags: tmi.ChatUserstate, requiredPermissions: Permission[]): boolean {
+        if(!requiredPermissions || requiredPermissions.length == 0 ) { // checks if requiredPermissions exists or if it has no requirements
+            return true
+        }
+
+
+        return requiredPermissions.some(permission =>{
+            switch(permission) {
+                case "broadcaster":
+                    return (tags.badges && tags.badges.broadcaster)
```

Comment:
```suggestion
                    return (tags.badges?.broadcaster)
```

this will make it more readable, and does the same thing. It will also avoid undefined errors

Reply:
Try to do the same thing for the other ones too